src/mysky/encrypted_files.ts

```diff
@@ -232,13 +232,17 @@ export function deriveEncryptedPathSeed(pathSeed: string, subPath: string, isDir
   validateString("subPath", subPath, "parameter");
   validateBoolean("isDirectory", isDirectory, "parameter");
 
-  // The path seed must be for a directory and not a file.
-  if (pathSeed.length !== ENCRYPTION_PATH_SEED_DIRECTORY_LENGTH) {
+  // The path seed must be for a directory or a MySky root path seed.
+  //
+  // NOTE: As a historical artifact, the root path seed is the same length as a
+  // file path seed.
+  const acceptedLengths = [ENCRYPTION_PATH_SEED_FILE_LENGTH, ENCRYPTION_PATH_SEED_DIRECTORY_LENGTH];
+  if (!acceptedLengths.includes(pathSeed.length)) {
```

Comment:
Is this what we want? This seems a bit strange to me but I might not be fully understanding what's going on here.
So a file path seed is 64 chars when hex-encoded, a directory path is 128, with the exception of the root path which can be 64 and thus equal to a file path seed...

Doesn't this then completely break the niceness of the solution where by the length of the seed we were able to tell without question what type it was? Also, aren't we breaking the validator in a way by adding the file length here as an accepted length?

Would it be more explicit if we were to add an `isRoot` parameter to this function?
Maybe it's nothing but I kind of have the feeling this commit is introducing a lot of weakness in the validator but I might be wrong, if so I apologize :D

Reply:
Yeah, very good point. Made a slight change to this after the latest discussions. See the last comment on this PR for a summary.